kse/src/org/kse/gui/actions/SignJwtAction.java

```diff
@@ -0,0 +1,170 @@
+package org.kse.gui.actions;
+
+import java.awt.Toolkit;
+import java.security.KeyStore;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.Provider;
+
+import javax.swing.ImageIcon;
+
+import org.kse.crypto.Password;
+import org.kse.crypto.keypair.KeyPairType;
+import org.kse.crypto.keypair.KeyPairUtil;
+import org.kse.crypto.signing.SignatureType;
+import org.kse.gui.KseFrame;
+import org.kse.gui.dialogs.DViewJwt;
+import org.kse.gui.dialogs.sign.CustomClaim;
+import org.kse.gui.dialogs.sign.DSignJwt;
+import org.kse.gui.error.DError;
+import org.kse.utilities.history.KeyStoreHistory;
+import org.kse.utilities.history.KeyStoreState;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTClaimsSet.Builder;
+import com.nimbusds.jwt.SignedJWT;
+
+/**
+ * Action to create a JWT (JSON Web Token)
+ *
+ */
+public class SignJwtAction extends KeyStoreExplorerAction {
+
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * Construct action.
+	 * 
+	 * @param kseFrame KeyStore Explorer frame
+	 */
+	public SignJwtAction(KseFrame kseFrame) {
+		super(kseFrame);
+		putValue(LONG_DESCRIPTION, res.getString("SignJwtAction.statusbar"));
+		putValue(NAME, res.getString("SignJwtAction.text"));
+		putValue(SHORT_DESCRIPTION, res.getString("SignJwtAction.tooltip"));
+		putValue(SMALL_ICON,
+				new ImageIcon(Toolkit.getDefaultToolkit().createImage(getClass().getResource("images/signcrl.png"))));
+	}
+
+	@Override
+	protected void doAction() {
+		try {
+			KeyStoreHistory history = kseFrame.getActiveKeyStoreHistory();
+			KeyStoreState currentState = history.getCurrentState();
+
+			String alias = kseFrame.getSelectedEntryAlias();
+
+			Password password = getEntryPassword(alias, currentState);
+
+			if (password == null) {
+				return;
+			}
+			KeyStore keyStore = currentState.getKeyStore();
+
+			Provider provider = null;
+			if (history.getExplicitProvider() != null) {
+				provider = history.getExplicitProvider();
+			}
```

Comment:
I think this can be simplified to: Provider provider = history.getExplicitProvider();

Reply:
Fixed